src/v/raft/consensus.cc

```diff
@@ -1728,6 +1732,12 @@ ss::future<> consensus::truncate_to_latest_snapshot() {
           return _log.truncate_prefix(storage::truncate_prefix_config(
             details::next_offset(_last_snapshot_index),
             _scheduling.default_iopc));
+      })
+      .then([this] {
+          // when log was prefix truncate flushed offset should be equal to at
+          // least log start offset.
+          _flushed_offset = std::max(
+            _log.offsets().start_offset, _flushed_offset);
```

Comment:
This looks suspicious to me - it implies that the `start_offset` itself is always flushed, which may not be the case?  I think `_last_snapshot_index` is more natural here - we know that the snapshot offset is always flushed because it is snapshotted.

Reply:
after prefix truncate `start_offset` is always flushed, i may change it if this is more confusing

Reply:
> after prefix truncate start_offset is always flushed

I was imagining some scenario like:
* there is a segment (not flushed) starting at offset 10
* snapshot with `_last_snapshot_index` = 9 arrives
* we prefix-truncate the log, `_log.offsets().start_offset` is now 10

why is 10 now flushed?

Also, there is a case when existing segment starts at say offset 5 and ends at offset > 9, and snapshot with `_last_snapshot_index` = 9 arrives. In this case the log start offset will remain at 5 and I think we can bump flushed offset to 9 instead of leaving it at 5.

Reply:
agreed, i will change that to `_last_snapshot_index`